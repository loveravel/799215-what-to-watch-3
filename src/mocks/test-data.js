export const films = [
  {
    id: 1,
    name: `Movie 1`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: true
  },
  {
    id: 2,
    name: `Movie 2`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideoLink: ``,
    video: ``,
    isFavorite: false
  },
  {
    id: 3,
    name: `Movie 3`,
    genre: `Drama`,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: false
  },
];

export const movie = {
  id: 1,
  name: `Movie 1`,
  genre: ``,
  released: 2019,
  backgroundImage: ``,
  backgroundColor: ``,
  previewImage: ``,
  posterImage: ``,
  description: ``,
  rating: 4.5,
  scoresCount: 300,
  runTime: 3,
  director: ``,
  starring: [],
  previewVideo: ``,
  video: ``,
  isFavorite: false
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const userData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};
