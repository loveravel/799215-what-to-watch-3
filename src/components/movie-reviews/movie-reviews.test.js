import React from "react";
import renderer from "react-test-renderer";

import {reviews} from "../../mocks/test-data.js";

import MovieReviews from "./movie-reviews.jsx";

it(`<MovieReviews /> is rendered correctly`, () => {
  const props = {
    reviews,
  };

  const tree = renderer
    .create(<MovieReviews {...props}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
