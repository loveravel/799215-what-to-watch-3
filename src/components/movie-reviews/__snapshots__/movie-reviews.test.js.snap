// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieReviews /> is rendered correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.5
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.5
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            May 8, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5.5
      </div>
    </div>
  </div>
</div>
`;
